src/redux/books/books.js

```diff
@@ -1,26 +1,42 @@
-import { createReducer } from '@reduxjs/toolkit';
-
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
 
-export const addBook = (payload) => ({
+const id = () => Math.floor((1 + Math.random()) * 0x10000)
+  .toString(16)
+  .substring(1);
+
+export const addBook = (book) => ({
   type: ADD_BOOK,
-  payload,
+  book,
 });
 
-export const removeBook = (payload) => ({
+export const removeBook = (itemId) => ({
   type: REMOVE_BOOK,
-  payload,
+  itemId,
 });
 
-const booksReducer = createReducer([], {
-  [ADD_BOOK]: (state, action) => {
-    state.push(action.payload);
+const booksReducer = (state = [
+  {
+    itemId: id(), title: 'Oliver Twist', author: 'Charles Dikens', category: 'sc-fi',
   },
-
-  [REMOVE_BOOK]: (state, action) => {
-    state.filter((payload) => payload.id !== action.payload.id);
+  {
+    itemId: id(), title: 'The World as Will and Representation', author: 'Arthur Schopenhauer', category: 'philosophy',
   },
-});
+  {
+    itemId: id(), title: 'Thus Spoke Zarathustra', author: 'Friedrich Nietzsche', category: 'philosophy',
+  },
+  {
+    itemId: id(), title: 'Les Misérables', author: 'Victor Hugo', category: 'historical fiction',
+  },
```

Comment:
- [ ] Kindly create a variable called `initialState` and assign your array of objects to it. You can then reference to it inside your booksReducer like this :point_down: 

```
    const booksReducer = (state = initialState, action) => {
         switch (action.type) {
          ........
          }
     }
``` 
The reason for this is to make your code clean and to follow best practices.

Reply:
done